include/usb/usb_device.h

```diff
@@ -74,6 +75,17 @@ extern "C" {
 
 #define MAX_PACKET_SIZE0    64        /**< maximum packet size for EP 0 */
 
+#if defined(DT_USBHS_MAXIMUM_SPEED) && \
```

Comment:
I think doing it this way is not good, readability suffers and worse, we need to touch it again to support more then one device controller, e.g. a SoC may have one FS controller and one HS controller. That why I want to check it [explicit ](https://github.com/zephyrproject-rtos/zephyr/pull/15147/commits/9657a10e2bedf655f5b27445ef39c8147e0442c8)for at least HS and FS.

Reply:
To support several device controllers we need to rewrite a lot anyway, so it is probably better to touch less code. And we do compare against one global `DT_USBHS_MAXIMUM_SPEED` which does not depends on the controller. Probably that check should be done in `usb_dc_***` function.

Reply:
3c750df3ddf6ba464389b49f69273a35abfbb118 make usb device controller check